### PR TITLE
`tsh kube login` by name prefix, labels, and/or query

### PR DESCRIPTION
**Pull request by GavinFrazar**
This PR allows a user to select the current context by prefix name during `tsh kube login`, and to filter the kube clusters for login by `--labels` and/or predicate `--query` - these flags are not allowed with `--all`.

changelog: `tsh kube login` now supports `--query` and `--labels` optional arguments.

Part of the implementation for [RFD 129](https://github.com/gravitational/teleport/blob/master/rfd/0129-discovery-name-templating.md).

As part of the changes, I also reduced the number of API calls to list kube clusters from 3 to one - in the case of a relogin required, or an access request is auto-generated, we of course still incur unavoidable extra API calls.

## Example
```
$ tsh kube ls
Kube Cluster Name Labels                                                         Selected
----------------- -------------------------------------------------------------- --------
test              account-id=123456789012,env=fake,origin=gavin,region=us-west-1
test-cluster      account-id=123456789012,origin=gavin,region=us-west-1

$ tsh kube login --labels env=fake
Logged into Kubernetes clusters:
test

Select a context and try 'kubectl version' to test the connection.

$ tsh kube login --query 'labels["region"] == "us-west-1" && labels["env"] != "fake"'
Logged into Kubernetes clusters:
test-cluster-eks-us-west-1-123456789012

Select a context and try 'kubectl version' to test the connection.

$ tsh kube ls
Kube Cluster Name Labels                                                         Selected
----------------- -------------------------------------------------------------- --------
test              account-id=123456789012,env=fake,origin=gavin,region=us-west-1
test-cluster      account-id=123456789012,origin=gavin,region=us-west-1

$ tsh kube login test-clus
Logged into Kubernetes cluster "test-cluster-eks-us-west-1-123456789012". Try 'kubectl version' to test the connection.

$ tsh kube login test
Logged into Kubernetes cluster "test". Try 'kubectl version' to test the connection.

$ tsh kube ls
Kube Cluster Name Labels                                                         Selected
----------------- -------------------------------------------------------------- --------
test              account-id=123456789012,env=fake,origin=gavin,region=us-west-1 *
test-cluster      account-id=123456789012,origin=gavin,region=us-west-1

$ tsh kube login test-clus
Logged into Kubernetes cluster "test-cluster-eks-us-west-1-123456789012". Try 'kubectl version' to test the connection.

$ tsh kube ls
Kube Cluster Name Labels                                                         Selected
----------------- -------------------------------------------------------------- --------
test              account-id=123456789012,env=fake,origin=gavin,region=us-west-1
test-cluster      account-id=123456789012,origin=gavin,region=us-west-1          *

$ tsh kube login te
ERROR: kubernetes cluster "te" matches multiple kubernetes clusters:

Kube Cluster Name                       Labels                                                                                                                                                                                               Selected
--------------------------------------- ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- --------
test                                    account-id=123456789012,env=fake,origin=gavin,region=us-west-1,teleport.dev/cloud=AWS,teleport.dev/origin=dynamic
test-cluster-eks-us-west-1-123456789012 account-id=123456789012,origin=gavin,region=us-west-1,teleport.dev/cloud=AWS,teleport.dev/origin=cloud,teleport.internal/discovered-name=test-cluster,teleport.internal/discovery-group-name=aws-dev

Hint: use 'tsh kube ls -v' or 'tsh kube ls --format=[json|yaml]' to list all kubernetes clusters with full details.
Hint: try selecting the kubernetes cluster with a more specific name (ex: tsh kube login test).
Hint: try selecting the kubernetes cluster with additional --labels or --query predicate.
```

Related issue:
- https://github.com/gravitational/teleport/issues/22438

### CODE REVIEW

File: tool/tsh/common/kube.go [hunk 1]
```diff
@@ -1588,3 +1666,24 @@ func accessRequestForKubeCluster(ctx context.Context, cf *CLIConf, tc *client.Te
 
 	return req, nil
 }
+
+// formatAmbiguousKubeCluster is a helper func that formats an ambiguous kube
+// cluster error message.
+func formatAmbiguousKubeCluster(cf *CLIConf, selectors resourceSelectors, kubeClusters types.KubeClusters) string {
+	// dont mark the selected cluster
```
Comment (smallinsky) [inline tool/tsh/common/kube.go:1673]:
```suggestion
	// don't mark the selected cluster
```

**Review by smallinsky — APPROVED** [review]
I have tested with this with GCP audto-discover and it works fine. But wonder if we can delay backporting to v13 resource name collision CLI changes after we will finish v13 test plan regression. 

@GavinFrazar 
Additionally we have also the `tsh proxy kube` command that needs to be extended.

File: tool/tsh/common/kube.go (hunk 1 above)
Comment (GavinFrazar) [reply tool/tsh/common/kube.go:1673]: > But wonder if we can delay backporting to v13 resource name collision CLI changes after we will finish v13 test plan regression.

I agree, and I've been delaying backporting these UX changes until the testplan so we avoid backporting regressions like what happened with my `tsh db` ux backport a little while ago

Comment (GavinFrazar) [reply tool/tsh/common/kube.go:1673]: >Additionally we have also the tsh proxy kube command that needs to be extended.

Separate PR will be coming for that, thanks for pointing it out

**Review by tigrato — APPROVED** [review]

**Pull request merged**